<a href="https://colab.research.google.com/github/Ryong1998/snowball/blob/main/3_Data_Preparation.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 1. 데이터 선택

- After_raw 폴더에 있는 파일들 중에서 사용할 레코드와 컬럼드를 선택하여 진행 할 예정
- 2_Data_Understanding.inpynb 파일을 참조

## 1-1. df_etf_base_info.pkl

### 1-1-1. 데이터 불러오기 및 확인

In [5]:
import pandas as pd

df_etf_base_info = pd.read_pickle('/content/drive/MyDrive/snowball/data/After_raw/df_etf_base_info.pkl')
print(df_etf_base_info.info())
df_etf_base_info.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 292 entries, 0 to 291
Data columns (total 5 columns):
 #   Column  Non-Null Count  Dtype         
---  ------  --------------  -----         
 0   단축코드    292 non-null    int64         
 1   한글종목약명  292 non-null    object        
 2   상장일     292 non-null    datetime64[ns]
 3   운용사     292 non-null    object        
 4   CU수량    292 non-null    int64         
dtypes: datetime64[ns](1), int64(2), object(2)
memory usage: 11.5+ KB
None


단축코드         한글종목약명        상장일       운용사   CU수량
0  445690    BNK 주주가치액티브 2022-10-27  비엔케이자산운용  50000
1  442260  마이티 다이나믹퀀트액티브 2022-10-25    디비자산운용  40000
2  159800     마이티 코스피100 2012-07-05    디비자산운용  40000
3  361580   KBSTAR 200TR 2020-08-21   케이비자산운용  50000
4  285000   KBSTAR 200IT 2017-12-08   케이비자산운용  20000

### 1-1-2. 컬럼 제거

- '상장일'은 추후 etf 가격 정보를 사용할 것이므로 굳이 필요가 없고, '운용사' 컬럼은 추후 사용할 필요가 없음으로 두개의 컬럼은 제거해도 무방

In [6]:
df_etf_base_info = df_etf_base_info[['단축코드','한글종목약명','CU수량']]
df_etf_base_info

단축코드            한글종목약명   CU수량
0    445690       BNK 주주가치액티브  50000
1    442260     마이티 다이나믹퀀트액티브  40000
2    159800        마이티 코스피100  40000
3    361580      KBSTAR 200TR  50000
4    285000      KBSTAR 200IT  20000
..      ...               ...    ...
287  376250  ARIRANG 탄소효율그린뉴딜  50000
288  433250   UNICORN R&D 액티브  50000
289  215620      HK S&P코리아로우볼  30000
290  391670      HK 베스트일레븐액티브  50000
291  391680         HK 하이볼액티브  50000

[292 rows x 3 columns]

## 1-2. df_pdf_files.pkl

### 1-2-1. 데이터 불러오기 및 확인

In [53]:
import pandas as pd

df_pdf_files = pd.read_pickle('/content/drive/MyDrive/snowball/data/After_raw/df_pdf_files.pkl')
print(df_pdf_files.info())
print(df_pdf_files.isnull().sum())
df_pdf_files.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2536672 entries, 0 to 2536671
Data columns (total 8 columns):
 #   Column     Dtype         
---  ------     -----         
 0   종목코드       object        
 1   구성종목명      object        
 2   주식수(계약수)   float64       
 3   평가금액       float64       
 4   시가총액       float64       
 5   시가총액 구성비중  float64       
 6   etf_code   int64         
 7   pdf_date   datetime64[ns]
dtypes: datetime64[ns](1), float64(4), int64(1), object(2)
memory usage: 154.8+ MB
None
종목코드              0
구성종목명            35
주식수(계약수)      98782
평가금액         680957
시가총액         641151
시가총액 구성비중    783584
etf_code          0
pdf_date          0
dtype: int64


종목코드  구성종목명  주식수(계약수)  평가금액         시가총액  시가총액 구성비중  etf_code   pdf_date
0  005930   삼성전자     246.0   NaN  144156000.0        NaN    100910 2008-02-01
1  005490  POSCO     174.0   NaN   88914000.0        NaN    100910 2008-02-01
2  060000   국민은행     749.0   NaN   45988600.0        NaN    100910 2008-02-01
3  055550   신한지주     840.0   NaN   42588000.0        NaN    100910 2008-02-01
4  009540  현대중공업     116.0   NaN   37758000.0        NaN    100910 2008-02-01

### 1-2-2. 컬럼 제거

- '주식수(계약수)' 컬럼과 '시가총액 구성비중' 컬럼은 추후 사용을 하지 않기에 제거

In [54]:
df_pdf_files = df_pdf_files.drop(columns=['주식수(계약수)', '시가총액 구성비중'])
df_pdf_files.head()

종목코드  구성종목명  평가금액         시가총액  etf_code   pdf_date
0  005930   삼성전자   NaN  144156000.0    100910 2008-02-01
1  005490  POSCO   NaN   88914000.0    100910 2008-02-01
2  060000   국민은행   NaN   45988600.0    100910 2008-02-01
3  055550   신한지주   NaN   42588000.0    100910 2008-02-01
4  009540  현대중공업   NaN   37758000.0    100910 2008-02-01

- 시가총액 또는 평가금액 둘 중 하나에 존재하는 null 값들을 채울 때 시가총액 -> 평가금액 순으로 우선순위를 두어서 금액을 판단하기로 함

In [55]:
# df_pdf_files에서 '시가총액'이 null인 행을 찾아서 '평가금액' 값으로 채우기
df_pdf_files['시가총액'].fillna(df_pdf_files['평가금액'], inplace=True)
print(df_pdf_files.isnull().sum())

종목코드             0
구성종목명           35
평가금액        680957
시가총액        572275
etf_code         0
pdf_date         0
dtype: int64


In [56]:
len(df_pdf_files.loc[(df_pdf_files['평가금액'].isnull())&(df_pdf_files['시가총액'].isnull()),:])

572275

- '시가총액'의 null 값은 '평가금액'으로 채움
- 남아있는 '시가총액'의 null 값은 572275개, '평가금액'과 '시가총액'이 둘 다 null 값인 것은 572275개로 동일함을 확인

In [57]:
# 사용안하는 '평가금액' 컬럼 제거
df_pdf_files = df_pdf_files.drop(columns=['평가금액'])
df_pdf_files.head()

종목코드  구성종목명         시가총액  etf_code   pdf_date
0  005930   삼성전자  144156000.0    100910 2008-02-01
1  005490  POSCO   88914000.0    100910 2008-02-01
2  060000   국민은행   45988600.0    100910 2008-02-01
3  055550   신한지주   42588000.0    100910 2008-02-01
4  009540  현대중공업   37758000.0    100910 2008-02-01

### 1-2-3. 데이터 필터링

- df_etf_base_info의 '단축코드' 컬럼에 있는 값들로 df_pdf_files의 'etf_code' 컬럼의 값들을 필터링

In [58]:
# df_etf_base_info의 '단축코드' 컬럼에 있는 값들을 리스트
etf_code_list = list(df_etf_base_info['단축코드'].unique())
etf_code_list[:5]

[445690, 442260, 159800, 361580, 285000]

In [59]:
df_pdf_files = df_pdf_files.loc[df_pdf_files['etf_code'].isin(etf_code_list),:]
print(df_pdf_files.info())
print(df_pdf_files.isnull().sum())
df_pdf_files.head()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1786753 entries, 0 to 2528341
Data columns (total 5 columns):
 #   Column    Dtype         
---  ------    -----         
 0   종목코드      object        
 1   구성종목명     object        
 2   시가총액      float64       
 3   etf_code  int64         
 4   pdf_date  datetime64[ns]
dtypes: datetime64[ns](1), float64(1), int64(1), object(2)
memory usage: 81.8+ MB
None
종목코드        0
구성종목명       0
시가총액        0
etf_code    0
pdf_date    0
dtype: int64


종목코드  구성종목명         시가총액  etf_code   pdf_date
0  005930   삼성전자  144156000.0    100910 2008-02-01
1  005490  POSCO   88914000.0    100910 2008-02-01
2  060000   국민은행   45988600.0    100910 2008-02-01
3  055550   신한지주   42588000.0    100910 2008-02-01
4  009540  현대중공업   37758000.0    100910 2008-02-01

- etf_code_list 의 'etf_code' 값들로 필터링을 했을 때, df_pdf_files 내의 null 값들이 제거됨을 확인

### 1-2-4. 데이터 값 변환

- '종목코드' 컬럼의 값들 중 사용하는 '종목코드'들은 6자릿수 임으로 '종목코드'가 6자리가 아닌 '종목코드'들을 확인 진행

In [60]:
# '종목코드'가 6자리수가 아닌 종목코드들을 확인
etf_stock_code_list = list(df_pdf_files['종목코드'].unique())
cash_code_list = list()
for etf_stock_code in etf_stock_code_list:
  if len(etf_stock_code)!=6:
    cash_code_list.append(etf_stock_code)
    print(etf_stock_code, df_pdf_files.loc[df_pdf_files['종목코드']==etf_stock_code,'구성종목명'].iloc[0])

KRD010010001 원화예금
KR41F6S90008 카카오뱅크 F 202209 (  10)
KR41F6SA0003 카카오뱅크 F 202210 (  10)
KR41F6SB0002 카카오뱅크 F 202211 (  10)
KR41F6SC0001 카카오뱅크 F 202212 (  10)
KR41F6T10004 카카오뱅크 F 202301 (  10)
KR4101T30009 코스피200 F 202303
KR41F6T20003 카카오뱅크 F 202302 (  10)
KR41F6T30002 카카오뱅크 F 202303 (  10)
KR4201F32770 코스피200 C 201103 277.5
CASH00000001 설정현금액
KR4201F42720 코스피200 C 201104 272.5
KR4201F52976 코스피200 C 201105 297.5
KR4201F62959 코스피200 C 201106 295.0
KR4201F72859 코스피200 C 201107 285.0
KR4201F82924 코스피200 C 201108 292.5
KR4201F92451 코스피200 C 201109 245.0
KR4201FA2506 코스피200 C 201110 250.0
KR4201FB2505 코스피200 C 201111 250.0
KR4201FC2470 코스피200 C 201112 247.5
KR4201G12622 코스피200 C 201201 262.5
KR4201G22555 코스피200 C 201202 255.0
KR4201G32778 코스피200 C 201203 277.5
KR4201G42751 코스피200 C 201204 275.0
KR4201G52776 코스피200 C 201205 277.5
KR4201G62700 코스피200 C 201206 270.0
KR4201G72600 코스피200 C 201207 260.0
KR4201G82476 코스피200 C 201208 247.5
KR4201G92707 코스피200 C 201209 270.0
KR4201GA2703 코스피200 C 20

In [61]:
cash_code_list[:5]

['KRD010010001',
 'KR41F6S90008',
 'KR41F6SA0003',
 'KR41F6SB0002',
 'KR41F6SC0001']

- 'https://m.blog.naver.com/PostView.naver?blogId=ej5824486&logNo=222544889198&categoryNo=211&proxyReferer=' 사이트를 통해,  'CASH00000001 설정현금액' 은 펀드의 설정 및 환매를 위한 현금 보유량임으로, 실질적인 주식투자 운용과는 별개로 운용사에서 펀드 관리를 위해 설정해 놓은 부분 임으로 CASH00000001 는 제거
- 'KRD010010001 원화예금' 는 원화예금을 나타내지만 다른 6자리가 아닌 '종목코드'들은 펀드등 으로 추후 가치평가를 하기 어려움, 그냥 '현금'으로 생각
-  'KRD010010001 원화예금' 의 '종목코드'는 '000000'으로, 다른 6자리가 아닌 '종목코드'들은 '999999'로 대체

In [62]:
# 종목코드 ''CASH00000001'를 제거
df_pdf_files = df_pdf_files.loc[df_pdf_files['종목코드']!='CASH00000001',:]
df_pdf_files

종목코드   구성종목명         시가총액  etf_code   pdf_date
0              005930    삼성전자  144156000.0    100910 2008-02-01
1              005490   POSCO   88914000.0    100910 2008-02-01
2              060000    국민은행   45988600.0    100910 2008-02-01
3              055550    신한지주   42588000.0    100910 2008-02-01
4              009540   현대중공업   37758000.0    100910 2008-02-01
...               ...     ...          ...       ...        ...
2528337        037560  LG헬로비전    8316440.0     98560 2023-03-02
2528338        006490    인스코비    7063842.0     98560 2023-03-02
2528339        122450      KX    5316150.0     98560 2023-03-02
2528340        126560   현대퓨처넷    5201250.0     98560 2023-03-02
2528341  KRD010010001    원화현금    5521454.0     98560 2023-03-02

[1786212 rows x 5 columns]

In [63]:
# 종목코드 'KRD010010001' 을 확인
df_pdf_files.loc[df_pdf_files['종목코드']=='KRD010010001',:]

종목코드 구성종목명       시가총액  etf_code   pdf_date
100      KRD010010001  원화예금  4177622.0    100910 2008-02-01
201      KRD010010001  원화예금  4088920.0    100910 2008-03-03
302      KRD010010001  원화예금  5488388.0    100910 2008-04-01
403      KRD010010001  원화예금  6231776.0    100910 2008-05-02
504      KRD010010001  원화예금  8110493.0    100910 2008-06-02
...               ...   ...        ...       ...        ...
2528301  KRD010010001  원화현금   469401.0     98560 2022-11-01
2528311  KRD010010001  원화현금   614644.0     98560 2022-12-01
2528321  KRD010010001  원화현금  5805927.0     98560 2023-01-02
2528331  KRD010010001  원화현금  5730149.0     98560 2023-02-01
2528341  KRD010010001  원화현금  5521454.0     98560 2023-03-02

[19620 rows x 5 columns]

In [64]:
# 종목코드 'KRD010010001' 을 변환
df_pdf_files.loc[df_pdf_files['종목코드']=='KRD010010001','종목코드'] = '000000'
df_pdf_files.loc[df_pdf_files['종목코드']=='000000',:]

종목코드 구성종목명       시가총액  etf_code   pdf_date
100      000000  원화예금  4177622.0    100910 2008-02-01
201      000000  원화예금  4088920.0    100910 2008-03-03
302      000000  원화예금  5488388.0    100910 2008-04-01
403      000000  원화예금  6231776.0    100910 2008-05-02
504      000000  원화예금  8110493.0    100910 2008-06-02
...         ...   ...        ...       ...        ...
2528301  000000  원화현금   469401.0     98560 2022-11-01
2528311  000000  원화현금   614644.0     98560 2022-12-01
2528321  000000  원화현금  5805927.0     98560 2023-01-02
2528331  000000  원화현금  5730149.0     98560 2023-02-01
2528341  000000  원화현금  5521454.0     98560 2023-03-02

[19620 rows x 5 columns]

In [65]:
# 종목코드가 6자리가 아닌 데이터를 확인
df_pdf_files.loc[df_pdf_files['종목코드'].isin(cash_code_list) ,:]

종목코드                  구성종목명         시가총액  etf_code   pdf_date
125250   KR41F6S90008  카카오뱅크 F 202209 (  10)   14389500.0    105190 2022-09-01
125450   KR41F6SA0003  카카오뱅크 F 202210 (  10)   10547000.0    105190 2022-10-04
125650   KR41F6SB0002  카카오뱅크 F 202211 (  10)    9063000.0    105190 2022-11-01
125850   KR41F6SC0001  카카오뱅크 F 202212 (  10)   13170500.0    105190 2022-12-01
126050   KR41F6T10004  카카오뱅크 F 202301 (  10)   12773000.0    105190 2023-01-02
...               ...                    ...          ...       ...        ...
2411873  KR4101T30009        코스피200 F 202303   58570000.0    438740 2023-01-02
2412039  KR4101T30009        코스피200 F 202303  477975000.0    438740 2023-02-01
2412219  KR4101T30009        코스피200 F 202303   94740000.0    438740 2023-03-02
2423725  KR4101T30009        코스피200 F 202303   33458250.0    448570 2023-02-01
2423764  KR4101T30009        코스피200 F 202303   35527500.0    448570 2023-03-02

[1432 rows x 5 columns]

In [66]:
# 종목코드가 6자리가 아닌 데이터의 종목코드 수정 및 확인
df_pdf_files.loc[df_pdf_files['종목코드'].isin(cash_code_list) ,'종목코드'] = '999999'
df_pdf_files.loc[df_pdf_files['종목코드']=='999999',:] # 값들이 잘 변경됨을 확인

종목코드                  구성종목명         시가총액  etf_code   pdf_date
125250   999999  카카오뱅크 F 202209 (  10)   14389500.0    105190 2022-09-01
125450   999999  카카오뱅크 F 202210 (  10)   10547000.0    105190 2022-10-04
125650   999999  카카오뱅크 F 202211 (  10)    9063000.0    105190 2022-11-01
125850   999999  카카오뱅크 F 202212 (  10)   13170500.0    105190 2022-12-01
126050   999999  카카오뱅크 F 202301 (  10)   12773000.0    105190 2023-01-02
...         ...                    ...          ...       ...        ...
2411873  999999        코스피200 F 202303   58570000.0    438740 2023-01-02
2412039  999999        코스피200 F 202303  477975000.0    438740 2023-02-01
2412219  999999        코스피200 F 202303   94740000.0    438740 2023-03-02
2423725  999999        코스피200 F 202303   33458250.0    448570 2023-02-01
2423764  999999        코스피200 F 202303   35527500.0    448570 2023-03-02

[1432 rows x 5 columns]

In [67]:
df_pdf_files = df_pdf_files.reset_index(drop=True)
print(df_pdf_files.info())
print(df_pdf_files.isnull().sum())
df_pdf_files.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1786212 entries, 0 to 1786211
Data columns (total 5 columns):
 #   Column    Dtype         
---  ------    -----         
 0   종목코드      object        
 1   구성종목명     object        
 2   시가총액      float64       
 3   etf_code  int64         
 4   pdf_date  datetime64[ns]
dtypes: datetime64[ns](1), float64(1), int64(1), object(2)
memory usage: 68.1+ MB
None
종목코드        0
구성종목명       0
시가총액        0
etf_code    0
pdf_date    0
dtype: int64


종목코드  구성종목명         시가총액  etf_code   pdf_date
0  005930   삼성전자  144156000.0    100910 2008-02-01
1  005490  POSCO   88914000.0    100910 2008-02-01
2  060000   국민은행   45988600.0    100910 2008-02-01
3  055550   신한지주   42588000.0    100910 2008-02-01
4  009540  현대중공업   37758000.0    100910 2008-02-01

## 1-3. df_etf_closing_price.pkl

In [68]:
df_etf_closing_price = pd.read_pickle('/content/drive/MyDrive/snowball/data/After_raw/df_etf_closing_price.pkl')
print(df_etf_closing_price.info())
print(df_etf_closing_price.isnull().sum())
df_etf_closing_price.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 743878 entries, 0 to 743877
Data columns (total 5 columns):
 #   Column    Non-Null Count   Dtype         
---  ------    --------------   -----         
 0   종목코드      743878 non-null  int64         
 1   종목명       743878 non-null  object        
 2   종가        743878 non-null  int64         
 3   시가총액      743878 non-null  int64         
 4   etf_date  743878 non-null  datetime64[ns]
dtypes: datetime64[ns](1), int64(3), object(1)
memory usage: 28.4+ MB
None
종목코드        0
종목명         0
종가          0
시가총액        0
etf_date    0
dtype: int64


종목코드               종목명     종가          시가총액   etf_date
0  152100       ARIRANG 200  24775  422413750000 2015-01-02
1  141240    ARIRANG K100EW   9115   15495500000 2015-01-02
2  122090   ARIRANG KOSPI50  16005    2880900000 2015-01-02
3  141250  ARIRANG KRX100EW   8825    3530000000 2015-01-02
4  117740     ARIRANG LG그룹&  10765    1076500000 2015-01-02

## 1-4. df_stock_closing_price.pkl

In [69]:
df_stock_closing_price = pd.read_pickle('/content/drive/MyDrive/snowball/data/After_raw/df_stock_closing_price.pkl')
print(df_stock_closing_price.info())
print(df_stock_closing_price.isnull().sum())
df_stock_closing_price.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4660295 entries, 0 to 4660294
Data columns (total 5 columns):
 #   Column      Dtype         
---  ------      -----         
 0   종목코드        object        
 1   종목명         object        
 2   종가          int64         
 3   시가총액        float64       
 4   stock_date  datetime64[ns]
dtypes: datetime64[ns](1), float64(1), int64(1), object(2)
memory usage: 177.8+ MB
None
종목코드          0
종목명           0
종가            0
시가총액          0
stock_date    0
dtype: int64


종목코드    종목명     종가          시가총액 stock_date
0  060310     3S   2415  9.875483e+10 2015-01-02
1  068400  AJ렌터카  17000  3.764871e+11 2015-01-02
2  006840  AK홀딩스  85100  1.127367e+12 2015-01-02
3  054620  AP시스템   6190  1.449077e+11 2015-01-02
4  090470  AST젯텍   3120  4.554516e+10 2015-01-02

## 1-5. df_Dart_PL.pkl

### 1-5-1. 데이터 불러오기 및 확인

In [94]:
df_Dart_PL = pd.read_pickle('/content/drive/MyDrive/snowball/data/After_raw/df_Dart_PL.pkl')
print(df_Dart_PL.info())
print(df_Dart_PL.isnull().sum())
df_Dart_PL.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2834266 entries, 0 to 2834265
Data columns (total 15 columns):
 #   Column  Dtype 
---  ------  ----- 
 0   재무제표종류  object
 1   종목코드    object
 2   회사명     object
 3   시장구분    object
 4   업종      int64 
 5   업종명     object
 6   결산월     int64 
 7   결산기준일   object
 8   보고서종류   object
 9   통화      object
 10  항목코드    object
 11  항목명     object
 12  분기누적금액  object
 13  분기표시    int64 
 14  연도표시    int64 
dtypes: int64(4), object(11)
memory usage: 324.4+ MB
None
재무제표종류         0
종목코드           0
회사명            0
시장구분           0
업종             0
업종명            0
결산월            0
결산기준일          0
보고서종류          0
통화             0
항목코드           0
항목명           24
분기누적금액    311765
분기표시           0
연도표시           0
dtype: int64


재무제표종류      종목코드 회사명       시장구분   업종  \
0  포괄손익계산서, 기능별 분류(세후기타포괄손익) - 별도재무제표  [060310]  3S  코스닥시장상장법인  291   
1  포괄손익계산서, 기능별 분류(세후기타포괄손익) - 별도재무제표  [060310]  3S  코스닥시장상장법인  291   
2  포괄손익계산서, 기능별 분류(세후기타포괄손익) - 별도재무제표  [060310]  3S  코스닥시장상장법인  291   
3  포괄손익계산서, 기능별 분류(세후기타포괄손익) - 별도재무제표  [060310]  3S  코스닥시장상장법인  291   
4  포괄손익계산서, 기능별 분류(세후기타포괄손익) - 별도재무제표  [060310]  3S  코스닥시장상장법인  291   

             업종명  결산월       결산기준일  보고서종류   통화  \
0  일반 목적용 기계 제조업    3  2015-03-31  사업보고서  KRW   
1  일반 목적용 기계 제조업    3  2015-03-31  사업보고서  KRW   
2  일반 목적용 기계 제조업    3  2015-03-31  사업보고서  KRW   
3  일반 목적용 기계 제조업    3  2015-03-31  사업보고서  KRW   
4  일반 목적용 기계 제조업    3  2015-03-31  사업보고서  KRW   

                                             항목코드       항목명          분기누적금액  \
0                                    ifrs_Revenue   수익(매출액)  21,104,219,053   
1                                ifrs_CostOfSales      매출원가  19,807,461,698   
2                                ifrs_GrossProfit     매출총이익   1,296,757,355   
3  dart_TotalSellingGeneralAdministrativeExpenses   판매비와관리비   5,402,979,409   
4                        dart_OperatingIncomeLoss  영업이익(손실)  -4,106,222,054   

   분기표시  연도표시  
0     4  2015  
1     4  2015  
2     4  2015  
3     4  2015  
4     4  2015

### 1-5-2. 데이터 값 변환

#### 1-5-2-1. '재무제표종류' 값 변경

- '재무제표종류' 컬럼에서 ['손익계산서, 기능별 분류 - 별도재무제표', '손익계산서, 성격별 분류 - 별도재무제표'] 는 '별도손익계산서' 변환
- '재무제표종류' 컬럼에서 ['손익계산서, 성격별 분류 - 연결재무제표', '손익계산서, 기능별 분류 - 연결재무제표'] 는 '연결손익계산서' 변환
- '재무제표종류' 컬럼에서 ['포괄손익계산서, 성격별 분류(세후기타포괄손익) - 별도재무제표', '포괄손익계산서, 성격별 분류(세전기타포괄손익) - 별도재무제표(선택)', '포괄손익계산서(세후기타포괄손익) - 별도재무제표', '포괄손익계산서, 기능별 분류(세후기타포괄손익) - 별도재무제표', '포괄손익계산서, 기능별 분류(세전기타포괄손익) - 별도재무제표(선택)', '포괄손익계산서(세전기타포괄손익) - 별도재무제표(선택)'] 는 '포괄별도손익계산서' 변환
- '재무제표종류' 컬럼에서 ['포괄손익계산서, 성격별 분류(세후기타포괄손익) - 연결재무제표', '포괄손익계산서(세전기타포괄손익) - 연결재무제표(선택)', '포괄손익계산서, 성격별 분류(세전기타포괄손익) - 연결재무제표(선택)', '포괄손익계산서, 기능별 분류(세후기타포괄손익) - 연결재무제표', '포괄손익계산서, 기능별 분류(세전기타포괄손익) - 연결재무제표(선택)', '포괄손익계산서(세후기타포괄손익) - 연결재무제표'] 는 '포괄연결손익계산서' 변환

In [95]:
# 재무제표 종류의 값들을 변경
df_Dart_PL.loc[df_Dart_PL['재무제표종류'].isin(['손익계산서, 기능별 분류 - 별도재무제표', '손익계산서, 성격별 분류 - 별도재무제표']),'재무제표종류'] = '별도손익계산서'
df_Dart_PL.loc[df_Dart_PL['재무제표종류'].isin(['손익계산서, 성격별 분류 - 연결재무제표', '손익계산서, 기능별 분류 - 연결재무제표']),'재무제표종류'] = '연결손익계산서'
df_Dart_PL.loc[df_Dart_PL['재무제표종류'].isin( ['포괄손익계산서, 성격별 분류(세후기타포괄손익) - 별도재무제표', '포괄손익계산서, 성격별 분류(세전기타포괄손익) - 별도재무제표(선택)',
                                           '포괄손익계산서(세후기타포괄손익) - 별도재무제표', '포괄손익계산서, 기능별 분류(세후기타포괄손익) - 별도재무제표',
                                           '포괄손익계산서, 기능별 분류(세전기타포괄손익) - 별도재무제표(선택)', '포괄손익계산서(세전기타포괄손익) - 별도재무제표(선택)']),'재무제표종류'] =  '포괄별도손익계산서'
df_Dart_PL.loc[df_Dart_PL['재무제표종류'].isin( ['포괄손익계산서, 성격별 분류(세후기타포괄손익) - 연결재무제표', '포괄손익계산서(세전기타포괄손익) - 연결재무제표(선택)',
                                           '포괄손익계산서, 성격별 분류(세전기타포괄손익) - 연결재무제표(선택)', '포괄손익계산서, 기능별 분류(세후기타포괄손익) - 연결재무제표',
                                           '포괄손익계산서, 기능별 분류(세전기타포괄손익) - 연결재무제표(선택)', '포괄손익계산서(세후기타포괄손익) - 연결재무제표']),'재무제표종류'] = '포괄연결손익계산서'
df_Dart_PL.head()

재무제표종류      종목코드 회사명       시장구분   업종            업종명  결산월       결산기준일  \
0  포괄별도손익계산서  [060310]  3S  코스닥시장상장법인  291  일반 목적용 기계 제조업    3  2015-03-31   
1  포괄별도손익계산서  [060310]  3S  코스닥시장상장법인  291  일반 목적용 기계 제조업    3  2015-03-31   
2  포괄별도손익계산서  [060310]  3S  코스닥시장상장법인  291  일반 목적용 기계 제조업    3  2015-03-31   
3  포괄별도손익계산서  [060310]  3S  코스닥시장상장법인  291  일반 목적용 기계 제조업    3  2015-03-31   
4  포괄별도손익계산서  [060310]  3S  코스닥시장상장법인  291  일반 목적용 기계 제조업    3  2015-03-31   

   보고서종류   통화                                            항목코드       항목명  \
0  사업보고서  KRW                                    ifrs_Revenue   수익(매출액)   
1  사업보고서  KRW                                ifrs_CostOfSales      매출원가   
2  사업보고서  KRW                                ifrs_GrossProfit     매출총이익   
3  사업보고서  KRW  dart_TotalSellingGeneralAdministrativeExpenses   판매비와관리비   
4  사업보고서  KRW                        dart_OperatingIncomeLoss  영업이익(손실)   

           분기누적금액  분기표시  연도표시  
0  21,104,219,053     4  2015  
1  19,807,461,698     4  2015  
2   1,296,757,355     4  2015  
3   5,402,979,409     4  2015  
4  -4,106,222,054     4  2015

#### 1-5-2-2. '종목코드' 값 변경

In [96]:
df_Dart_PL['종목코드'] = df_Dart_PL['종목코드'].str.replace('[', '').str.replace(']', '')
df_Dart_PL.head()

<ipython-input-96-3f3328594445>:1: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  df_Dart_PL['종목코드'] = df_Dart_PL['종목코드'].str.replace('[', '').str.replace(']', '')


재무제표종류    종목코드 회사명       시장구분   업종            업종명  결산월       결산기준일  \
0  포괄별도손익계산서  060310  3S  코스닥시장상장법인  291  일반 목적용 기계 제조업    3  2015-03-31   
1  포괄별도손익계산서  060310  3S  코스닥시장상장법인  291  일반 목적용 기계 제조업    3  2015-03-31   
2  포괄별도손익계산서  060310  3S  코스닥시장상장법인  291  일반 목적용 기계 제조업    3  2015-03-31   
3  포괄별도손익계산서  060310  3S  코스닥시장상장법인  291  일반 목적용 기계 제조업    3  2015-03-31   
4  포괄별도손익계산서  060310  3S  코스닥시장상장법인  291  일반 목적용 기계 제조업    3  2015-03-31   

   보고서종류   통화                                            항목코드       항목명  \
0  사업보고서  KRW                                    ifrs_Revenue   수익(매출액)   
1  사업보고서  KRW                                ifrs_CostOfSales      매출원가   
2  사업보고서  KRW                                ifrs_GrossProfit     매출총이익   
3  사업보고서  KRW  dart_TotalSellingGeneralAdministrativeExpenses   판매비와관리비   
4  사업보고서  KRW                        dart_OperatingIncomeLoss  영업이익(손실)   

           분기누적금액  분기표시  연도표시  
0  21,104,219,053     4  2015  
1  19,807,461,698     4  2015  
2   1,296,757,355     4  2015  
3   5,402,979,409     4  2015  
4  -4,106,222,054     4  2015

In [97]:
# df_Dart_PL 에서 '종목코드'의 값들이 6임으로 확인
for stock_code in df_Dart_PL['종목코드'].unique():
  if len(stock_code) !=6:
    print(stock_code)

#### 1-5-2-3. '분기누적금액' 값 변경

- '분기누적금액'은 object 타입임으로, int 타입으로 변경해야 추후 수치계산이 가능
- int 타입으로 변경한후, '통화' 컬럼의 값을 반영해서 '분기누적금액'의 통화 단위를 '원'으로 통일

In [98]:
# '분기누적금액' 컬럼 타입 변경
df_Dart_PL['분기누적금액'] = df_Dart_PL['분기누적금액'].str.replace(',', '')

# '분기누적금액' 컬럼의 NaN 값은 그대로 두고, 숫자 형태의 값만 정수로 변환
df_Dart_PL['분기누적금액'] = pd.to_numeric(df_Dart_PL['분기누적금액'], errors='coerce')


print(df_Dart_PL.info())
print(df_Dart_PL.isnull().sum())
df_Dart_PL.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2834266 entries, 0 to 2834265
Data columns (total 15 columns):
 #   Column  Dtype  
---  ------  -----  
 0   재무제표종류  object 
 1   종목코드    object 
 2   회사명     object 
 3   시장구분    object 
 4   업종      int64  
 5   업종명     object 
 6   결산월     int64  
 7   결산기준일   object 
 8   보고서종류   object 
 9   통화      object 
 10  항목코드    object 
 11  항목명     object 
 12  분기누적금액  float64
 13  분기표시    int64  
 14  연도표시    int64  
dtypes: float64(1), int64(4), object(10)
memory usage: 324.4+ MB
None
재무제표종류         0
종목코드           0
회사명            0
시장구분           0
업종             0
업종명            0
결산월            0
결산기준일          0
보고서종류          0
통화             0
항목코드           0
항목명           24
분기누적금액    311765
분기표시           0
연도표시           0
dtype: int64


재무제표종류    종목코드 회사명       시장구분   업종            업종명  결산월       결산기준일  \
0  포괄별도손익계산서  060310  3S  코스닥시장상장법인  291  일반 목적용 기계 제조업    3  2015-03-31   
1  포괄별도손익계산서  060310  3S  코스닥시장상장법인  291  일반 목적용 기계 제조업    3  2015-03-31   
2  포괄별도손익계산서  060310  3S  코스닥시장상장법인  291  일반 목적용 기계 제조업    3  2015-03-31   
3  포괄별도손익계산서  060310  3S  코스닥시장상장법인  291  일반 목적용 기계 제조업    3  2015-03-31   
4  포괄별도손익계산서  060310  3S  코스닥시장상장법인  291  일반 목적용 기계 제조업    3  2015-03-31   

   보고서종류   통화                                            항목코드       항목명  \
0  사업보고서  KRW                                    ifrs_Revenue   수익(매출액)   
1  사업보고서  KRW                                ifrs_CostOfSales      매출원가   
2  사업보고서  KRW                                ifrs_GrossProfit     매출총이익   
3  사업보고서  KRW  dart_TotalSellingGeneralAdministrativeExpenses   판매비와관리비   
4  사업보고서  KRW                        dart_OperatingIncomeLoss  영업이익(손실)   

         분기누적금액  분기표시  연도표시  
0  2.110422e+10     4  2015  
1  1.980746e+10     4  2015  
2  1.296757e+09     4  2015  
3  5.402979e+09     4  2015  
4 -4.106222e+09     4  2015

In [99]:
# 외환 정보 데이터 불러오기
df_exchange = pd.read_pickle('/content/drive/MyDrive/snowball/data/After_raw/df_exchange.pkl')
print(df_exchange.info())
df_exchange.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9060 entries, 0 to 9059
Data columns (total 8 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   날짜      9060 non-null   object
 1   종가      9060 non-null   object
 2   시가      9060 non-null   object
 3   고가      9060 non-null   object
 4   저가      9060 non-null   object
 5   거래량     1057 non-null   object
 6   변동 %    9060 non-null   object
 7   외환종류    9060 non-null   object
dtypes: object(8)
memory usage: 566.4+ KB
None


날짜        종가        시가        고가        저가  거래량    변동 % 외환종류
0  2023-09-06  1,329.76  1,331.62  1,337.57  1,329.14  NaN   0.05%  USD
1  2023-09-05  1,329.05  1,318.41  1,335.68  1,316.81  NaN   0.90%  USD
2  2023-09-04  1,317.21  1,318.45  1,322.00  1,315.59  NaN  -0.03%  USD
3  2023-09-01  1,317.55  1,325.46  1,326.37  1,311.88  NaN  -0.57%  USD
4  2023-08-31  1,325.14  1,323.79  1,326.86  1,320.00  NaN   0.12%  USD

In [100]:
df_exchange['종가'] = df_exchange['종가'].str.replace(',', '')
df_exchange['종가'] = df_exchange['종가'].astype(float)
print(df_exchange.info())
df_exchange.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9060 entries, 0 to 9059
Data columns (total 8 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   날짜      9060 non-null   object 
 1   종가      2265 non-null   float64
 2   시가      9060 non-null   object 
 3   고가      9060 non-null   object 
 4   저가      9060 non-null   object 
 5   거래량     1057 non-null   object 
 6   변동 %    9060 non-null   object 
 7   외환종류    9060 non-null   object 
dtypes: float64(1), object(7)
memory usage: 566.4+ KB
None


날짜       종가        시가        고가        저가  거래량    변동 % 외환종류
0  2023-09-06  1329.76  1,331.62  1,337.57  1,329.14  NaN   0.05%  USD
1  2023-09-05  1329.05  1,318.41  1,335.68  1,316.81  NaN   0.90%  USD
2  2023-09-04  1317.21  1,318.45  1,322.00  1,315.59  NaN  -0.03%  USD
3  2023-09-01  1317.55  1,325.46  1,326.37  1,311.88  NaN  -0.57%  USD
4  2023-08-31  1325.14  1,323.79  1,326.86  1,320.00  NaN   0.12%  USD

In [101]:
# merge를 통해서 데이터 병합
df_Dart_PL = pd.merge(df_Dart_PL, df_exchange[['날짜','종가','외환종류']], how='left', left_on = ['결산기준일','통화'], right_on=['날짜','외환종류'])
df_Dart_PL.head()

재무제표종류    종목코드 회사명       시장구분   업종            업종명  결산월       결산기준일  \
0  포괄별도손익계산서  060310  3S  코스닥시장상장법인  291  일반 목적용 기계 제조업    3  2015-03-31   
1  포괄별도손익계산서  060310  3S  코스닥시장상장법인  291  일반 목적용 기계 제조업    3  2015-03-31   
2  포괄별도손익계산서  060310  3S  코스닥시장상장법인  291  일반 목적용 기계 제조업    3  2015-03-31   
3  포괄별도손익계산서  060310  3S  코스닥시장상장법인  291  일반 목적용 기계 제조업    3  2015-03-31   
4  포괄별도손익계산서  060310  3S  코스닥시장상장법인  291  일반 목적용 기계 제조업    3  2015-03-31   

   보고서종류   통화                                            항목코드       항목명  \
0  사업보고서  KRW                                    ifrs_Revenue   수익(매출액)   
1  사업보고서  KRW                                ifrs_CostOfSales      매출원가   
2  사업보고서  KRW                                ifrs_GrossProfit     매출총이익   
3  사업보고서  KRW  dart_TotalSellingGeneralAdministrativeExpenses   판매비와관리비   
4  사업보고서  KRW                        dart_OperatingIncomeLoss  영업이익(손실)   

         분기누적금액  분기표시  연도표시   날짜  종가 외환종류  
0  2.110422e+10     4  2015  NaN NaN  NaN  
1  1.980746e+10     4  2015  NaN NaN  NaN  
2  1.296757e+09     4  2015  NaN NaN  NaN  
3  5.402979e+09     4  2015  NaN NaN  NaN  
4 -4.106222e+09     4  2015  NaN NaN  NaN

In [102]:
df_Dart_PL.loc[(df_Dart_PL['종가'].notnull()) & (df_Dart_PL['분기누적금액'].notnull()),:] # 수정 전 수정할 대상 값들을 확인

재무제표종류    종목코드         회사명        시장구분   업종               업종명  \
6051     포괄별도손익계산서  900100       뉴프라이드   코스닥시장상장법인  452  자동차 부품 및 내장품 판매업   
6052     포괄별도손익계산서  900100       뉴프라이드   코스닥시장상장법인  452  자동차 부품 및 내장품 판매업   
6053     포괄별도손익계산서  900100       뉴프라이드   코스닥시장상장법인  452  자동차 부품 및 내장품 판매업   
6054     포괄별도손익계산서  900100       뉴프라이드   코스닥시장상장법인  452  자동차 부품 및 내장품 판매업   
6055     포괄별도손익계산서  900100       뉴프라이드   코스닥시장상장법인  452  자동차 부품 및 내장품 판매업   
...            ...     ...         ...         ...  ...               ...   
2828479  포괄연결손익계산서  950210  프레스티지바이오파마  유가증권시장상장법인  701   자연과학 및 공학 연구개발업   
2828480  포괄연결손익계산서  950210  프레스티지바이오파마  유가증권시장상장법인  701   자연과학 및 공학 연구개발업   
2828481  포괄연결손익계산서  950210  프레스티지바이오파마  유가증권시장상장법인  701   자연과학 및 공학 연구개발업   
2828483  포괄연결손익계산서  950210  프레스티지바이오파마  유가증권시장상장법인  701   자연과학 및 공학 연구개발업   
2828484  포괄연결손익계산서  950210  프레스티지바이오파마  유가증권시장상장법인  701   자연과학 및 공학 연구개발업   

         결산월       결산기준일   보고서종류   통화  \
6051      12  2015-12-31   사업보고서  USD   
6052      12  2015-12-31   사업보고서  USD   
6053      12  2015-12-31   사업보고서  USD   
6054      12  2015-12-31   사업보고서  USD   
6055      12  2015-12-31   사업보고서  USD   
...      ...         ...     ...  ...   
2828479    6  2022-03-31  3분기보고서  USD   
2828480    6  2022-03-31  3분기보고서  USD   
2828481    6  2022-03-31  3분기보고서  USD   
2828483    6  2022-03-31  3분기보고서  USD   
2828484    6  2022-03-31  3분기보고서  USD   

                                                   항목코드        항목명  \
6051                                       ifrs_Revenue    수익(매출액)   
6052                                   ifrs_CostOfSales       매출원가   
6053                                   ifrs_GrossProfit      매출총이익   
6054     dart_TotalSellingGeneralAdministrativeExpenses    판매비와관리비   
6055                           dart_OperatingIncomeLoss   영업이익(손실)   
...                                                 ...        ...   
2828479                            ifrs-full_ProfitLoss      분기순손실   
2828480              ifrs-full_OtherComprehensiveIncome     기타포괄손실   
2828481                   ifrs-full_ComprehensiveIncome      총포괄손실   
2828483             ifrs-full_BasicEarningsLossPerShare     기본주당손실   
2828484           ifrs-full_DilutedEarningsLossPerShare     희석주당손실   

             분기누적금액  분기표시  연도표시          날짜       종가 외환종류  
6051     53519782.0     4  2015  2015-12-31  1175.95  USD  
6052     47973559.0     4  2015  2015-12-31  1175.95  USD  
6053      5546223.0     4  2015  2015-12-31  1175.95  USD  
6054      6572269.0     4  2015  2015-12-31  1175.95  USD  
6055     -1026046.0     4  2015  2015-12-31  1175.95  USD  
...             ...   ...   ...         ...      ...  ...  
2828479 -93223933.0     3  2022  2022-03-31  1214.17  USD  
2828480  -1507885.0     3  2022  2022-03-31  1214.17  USD  
2828481 -94731818.0     3  2022  2022-03-31  1214.17  USD  
2828483        -8.0     3  2022  2022-03-31  1214.17  USD  
2828484        -8.0     3  2022  2022-03-31  1214.17  USD  

[4186 rows x 18 columns]

In [103]:
df_Dart_PL.loc[(df_Dart_PL['종가'].notnull()) & (df_Dart_PL['분기누적금액'].notnull()),'분기누적금액'] = df_Dart_PL['분기누적금액'] * df_Dart_PL['종가'] # 환율을 고려해서 '분기누적금액'을 수정

In [104]:
df_Dart_PL.loc[6051] # 값이 잘 변경되었음 확인

재무제표종류             포괄별도손익계산서
종목코드                  900100
회사명                    뉴프라이드
시장구분               코스닥시장상장법인
업종                       452
업종명         자동차 부품 및 내장품 판매업
결산월                       12
결산기준일             2015-12-31
보고서종류                  사업보고서
통화                       USD
항목코드            ifrs_Revenue
항목명                  수익(매출액)
분기누적금액    62936587642.900002
분기표시                       4
연도표시                    2015
날짜                2015-12-31
종가                   1175.95
외환종류                     USD
Name: 6051, dtype: object

In [105]:
print(df_Dart_PL.info())
print(df_Dart_PL.isnull().sum())
df_Dart_PL.head()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2834266 entries, 0 to 2834265
Data columns (total 18 columns):
 #   Column  Dtype  
---  ------  -----  
 0   재무제표종류  object 
 1   종목코드    object 
 2   회사명     object 
 3   시장구분    object 
 4   업종      int64  
 5   업종명     object 
 6   결산월     int64  
 7   결산기준일   object 
 8   보고서종류   object 
 9   통화      object 
 10  항목코드    object 
 11  항목명     object 
 12  분기누적금액  float64
 13  분기표시    int64  
 14  연도표시    int64  
 15  날짜      object 
 16  종가      float64
 17  외환종류    object 
dtypes: float64(2), int64(4), object(12)
memory usage: 475.4+ MB
None
재무제표종류          0
종목코드            0
회사명             0
시장구분            0
업종              0
업종명             0
결산월             0
결산기준일           0
보고서종류           0
통화              0
항목코드            0
항목명            24
분기누적금액     311765
분기표시            0
연도표시            0
날짜        2822266
종가        2829523
외환종류      2822266
dtype: int64


재무제표종류    종목코드 회사명       시장구분   업종            업종명  결산월       결산기준일  \
0  포괄별도손익계산서  060310  3S  코스닥시장상장법인  291  일반 목적용 기계 제조업    3  2015-03-31   
1  포괄별도손익계산서  060310  3S  코스닥시장상장법인  291  일반 목적용 기계 제조업    3  2015-03-31   
2  포괄별도손익계산서  060310  3S  코스닥시장상장법인  291  일반 목적용 기계 제조업    3  2015-03-31   
3  포괄별도손익계산서  060310  3S  코스닥시장상장법인  291  일반 목적용 기계 제조업    3  2015-03-31   
4  포괄별도손익계산서  060310  3S  코스닥시장상장법인  291  일반 목적용 기계 제조업    3  2015-03-31   

   보고서종류   통화                                            항목코드       항목명  \
0  사업보고서  KRW                                    ifrs_Revenue   수익(매출액)   
1  사업보고서  KRW                                ifrs_CostOfSales      매출원가   
2  사업보고서  KRW                                ifrs_GrossProfit     매출총이익   
3  사업보고서  KRW  dart_TotalSellingGeneralAdministrativeExpenses   판매비와관리비   
4  사업보고서  KRW                        dart_OperatingIncomeLoss  영업이익(손실)   

         분기누적금액  분기표시  연도표시   날짜  종가 외환종류  
0  2.110422e+10     4  2015  NaN NaN  NaN  
1  1.980746e+10     4  2015  NaN NaN  NaN  
2  1.296757e+09     4  2015  NaN NaN  NaN  
3  5.402979e+09     4  2015  NaN NaN  NaN  
4 -4.106222e+09     4  2015  NaN NaN  NaN

#### 1-5-2-4. '항목명' 값 수정

In [106]:
df_Dart_PL.loc[df_Dart_PL['항목명'].isnull(),:]

재무제표종류    종목코드   회사명       시장구분   업종             업종명  결산월  \
20740   포괄별도손익계산서  061040  알에프텍  코스닥시장상장법인  264  통신 및 방송 장비 제조업   12   
20749   포괄별도손익계산서  061040  알에프텍  코스닥시장상장법인  264  통신 및 방송 장비 제조업   12   
69941   포괄연결손익계산서  061040  알에프텍  코스닥시장상장법인  264  통신 및 방송 장비 제조업   12   
69951   포괄연결손익계산서  061040  알에프텍  코스닥시장상장법인  264  통신 및 방송 장비 제조업   12   
69953   포괄연결손익계산서  061040  알에프텍  코스닥시장상장법인  264  통신 및 방송 장비 제조업   12   
69960   포괄연결손익계산서  061040  알에프텍  코스닥시장상장법인  264  통신 및 방송 장비 제조업   12   
110114  포괄별도손익계산서  061040  알에프텍  코스닥시장상장법인  264  통신 및 방송 장비 제조업   12   
110123  포괄별도손익계산서  061040  알에프텍  코스닥시장상장법인  264  통신 및 방송 장비 제조업   12   
151454  포괄연결손익계산서  061040  알에프텍  코스닥시장상장법인  264  통신 및 방송 장비 제조업   12   
151464  포괄연결손익계산서  061040  알에프텍  코스닥시장상장법인  264  통신 및 방송 장비 제조업   12   
151466  포괄연결손익계산서  061040  알에프텍  코스닥시장상장법인  264  통신 및 방송 장비 제조업   12   
151473  포괄연결손익계산서  061040  알에프텍  코스닥시장상장법인  264  통신 및 방송 장비 제조업   12   
193447  포괄연결손익계산서  061040  알에프텍  코스닥시장상장법인  264  통신 및 방송 장비 제조업   12   
193457  포괄연결손익계산서  061040  알에프텍  코스닥시장상장법인  264  통신 및 방송 장비 제조업   12   
193459  포괄연결손익계산서  061040  알에프텍  코스닥시장상장법인  264  통신 및 방송 장비 제조업   12   
193466  포괄연결손익계산서  061040  알에프텍  코스닥시장상장법인  264  통신 및 방송 장비 제조업   12   
307067  포괄별도손익계산서  061040  알에프텍  코스닥시장상장법인  264  통신 및 방송 장비 제조업   12   
307076  포괄별도손익계산서  061040  알에프텍  코스닥시장상장법인  264  통신 및 방송 장비 제조업   12   
347101  포괄별도손익계산서  061040  알에프텍  코스닥시장상장법인  264  통신 및 방송 장비 제조업   12   
347110  포괄별도손익계산서  061040  알에프텍  코스닥시장상장법인  264  통신 및 방송 장비 제조업   12   
431360  포괄연결손익계산서  061040  알에프텍  코스닥시장상장법인  264  통신 및 방송 장비 제조업   12   
431370  포괄연결손익계산서  061040  알에프텍  코스닥시장상장법인  264  통신 및 방송 장비 제조업   12   
431372  포괄연결손익계산서  061040  알에프텍  코스닥시장상장법인  264  통신 및 방송 장비 제조업   12   
431379  포괄연결손익계산서  061040  알에프텍  코스닥시장상장법인  264  통신 및 방송 장비 제조업   12   

             결산기준일   보고서종류   통화  \
20740   2015-12-31   사업보고서  KRW   
20749   2015-12-31   사업보고서  KRW   
69941   2015-12-31   사업보고서  KRW   
69951   2015-12-31   사업보고서  KRW   
69953   2015-12-31   사업보고서  KRW   
69960   2015-12-31   사업보고서  KRW   
110114  2016-09-30  3분기보고서  KRW   
110123  2016-09-30  3분기보고서  KRW   
151454  2016-09-30  3분기보고서  KRW   
151464  2016-09-30  3분기보고서  KRW   
151466  2016-09-30  3분기보고서  KRW   
151473  2016-09-30  3분기보고서  KRW   
193447  2016-06-30   반기보고서  KRW   
193457  2016-06-30   반기보고서  KRW   
193459  2016-06-30   반기보고서  KRW   
193466  2016-06-30   반기보고서  KRW   
307067  2016-06-30   반기보고서  KRW   
307076  2016-06-30   반기보고서  KRW   
347101  2016-03-31  1분기보고서  KRW   
347110  2016-03-31  1분기보고서  KRW   
431360  2016-03-31  1분기보고서  KRW   
431370  2016-03-31  1분기보고서  KRW   
431372  2016-03-31  1분기보고서  KRW   
431379  2016-03-31  1분기보고서  KRW   

                                                     항목코드  항목명  분기누적금액  분기표시  \
20740   entity00309831_udf_IS_2014325104537565_Stateme...  NaN     NaN     4   
20749   entity00309831_udf_IS_2014325104719772_Stateme...  NaN     NaN     4   
69941   entity00309831_udf_IS_2014314164958720_Stateme...  NaN     NaN     4   
69951   entity00309831_udf_IS_2014314165454905_Stateme...  NaN     NaN     4   
69953   entity00309831_udf_IS_20143141657323_Statement...  NaN     NaN     4   
69960   entity00309831_udf_IS_2014314165828706_Stateme...  NaN     NaN     4   
110114  entity00309831_udf_IS_2014325104537565_Stateme...  NaN     NaN     3   
110123  entity00309831_udf_IS_2014325104719772_Stateme...  NaN     NaN     3   
151454  entity00309831_udf_IS_2014314164958720_Stateme...  NaN     NaN     3   
151464  entity00309831_udf_IS_2014314165454905_Stateme...  NaN     NaN     3   
151466  entity00309831_udf_IS_20143141657323_Statement...  NaN     NaN     3   
151473  entity00309831_udf_IS_2014314165828706_Stateme...  NaN     NaN     3   
193447  entity00309831_udf_IS_2014314164958720_Stateme...  NaN     NaN     2   
193457  entity00309831_udf_IS_2014314165454905_Stateme...  NaN     NaN     2   
193459  entity00309831_udf_IS_20143141657323_Statement...  NaN     NaN     2   
193466  entity00309831_udf_IS_2014314165828706_Stateme

- 확인 결과 '항목명'이 null 값은 값들은 추후 사용에 적합한 데이터가 아님, 해당 값들 제거

In [107]:
df_Dart_PL = df_Dart_PL.loc[df_Dart_PL['항목명'].notnull(),:]
print(df_Dart_PL.info())
print(df_Dart_PL.isnull().sum())
df_Dart_PL.head()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2834242 entries, 0 to 2834265
Data columns (total 18 columns):
 #   Column  Dtype  
---  ------  -----  
 0   재무제표종류  object 
 1   종목코드    object 
 2   회사명     object 
 3   시장구분    object 
 4   업종      int64  
 5   업종명     object 
 6   결산월     int64  
 7   결산기준일   object 
 8   보고서종류   object 
 9   통화      object 
 10  항목코드    object 
 11  항목명     object 
 12  분기누적금액  float64
 13  분기표시    int64  
 14  연도표시    int64  
 15  날짜      object 
 16  종가      float64
 17  외환종류    object 
dtypes: float64(2), int64(4), object(12)
memory usage: 410.8+ MB
None
재무제표종류          0
종목코드            0
회사명             0
시장구분            0
업종              0
업종명             0
결산월             0
결산기준일           0
보고서종류           0
통화              0
항목코드            0
항목명             0
분기누적금액     311741
분기표시            0
연도표시            0
날짜        2822242
종가        2829499
외환종류      2822242
dtype: int64


재무제표종류    종목코드 회사명       시장구분   업종            업종명  결산월       결산기준일  \
0  포괄별도손익계산서  060310  3S  코스닥시장상장법인  291  일반 목적용 기계 제조업    3  2015-03-31   
1  포괄별도손익계산서  060310  3S  코스닥시장상장법인  291  일반 목적용 기계 제조업    3  2015-03-31   
2  포괄별도손익계산서  060310  3S  코스닥시장상장법인  291  일반 목적용 기계 제조업    3  2015-03-31   
3  포괄별도손익계산서  060310  3S  코스닥시장상장법인  291  일반 목적용 기계 제조업    3  2015-03-31   
4  포괄별도손익계산서  060310  3S  코스닥시장상장법인  291  일반 목적용 기계 제조업    3  2015-03-31   

   보고서종류   통화                                            항목코드       항목명  \
0  사업보고서  KRW                                    ifrs_Revenue   수익(매출액)   
1  사업보고서  KRW                                ifrs_CostOfSales      매출원가   
2  사업보고서  KRW                                ifrs_GrossProfit     매출총이익   
3  사업보고서  KRW  dart_TotalSellingGeneralAdministrativeExpenses   판매비와관리비   
4  사업보고서  KRW                        dart_OperatingIncomeLoss  영업이익(손실)   

         분기누적금액  분기표시  연도표시   날짜  종가 외환종류  
0  2.110422e+10     4  2015  NaN NaN  NaN  
1  1.980746e+10     4  2015  NaN NaN  NaN  
2  1.296757e+09     4  2015  NaN NaN  NaN  
3  5.402979e+09     4  2015  NaN NaN  NaN  
4 -4.106222e+09     4  2015  NaN NaN  NaN

### 1-5-3. 컬럼제거

- '시장구분', '업종', '업종명', '결산월', '보고서종류', '통화', '날짜', '종가', '외환종류' 컬럼들은 추후 사용을 안할 예정 임으로 제거

In [108]:
df_Dart_PL = df_Dart_PL[['재무제표종류', '종목코드', '결산기준일', '연도표시', '분기표시','항목코드', '항목명', '분기누적금액']]
df_Dart_PL.head()

재무제표종류    종목코드       결산기준일  연도표시  분기표시  \
0  포괄별도손익계산서  060310  2015-03-31  2015     4   
1  포괄별도손익계산서  060310  2015-03-31  2015     4   
2  포괄별도손익계산서  060310  2015-03-31  2015     4   
3  포괄별도손익계산서  060310  2015-03-31  2015     4   
4  포괄별도손익계산서  060310  2015-03-31  2015     4   

                                             항목코드       항목명        분기누적금액  
0                                    ifrs_Revenue   수익(매출액)  2.110422e+10  
1                                ifrs_CostOfSales      매출원가  1.980746e+10  
2                                ifrs_GrossProfit     매출총이익  1.296757e+09  
3  dart_TotalSellingGeneralAdministrativeExpenses   판매비와관리비  5.402979e+09  
4                        dart_OperatingIncomeLoss  영업이익(손실) -4.106222e+09

In [109]:
df_Dart_PL = df_Dart_PL.reset_index(drop=True)

print(df_Dart_PL.info())
print(df_Dart_PL.isnull().sum())
df_Dart_PL.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2834242 entries, 0 to 2834241
Data columns (total 8 columns):
 #   Column  Dtype  
---  ------  -----  
 0   재무제표종류  object 
 1   종목코드    object 
 2   결산기준일   object 
 3   연도표시    int64  
 4   분기표시    int64  
 5   항목코드    object 
 6   항목명     object 
 7   분기누적금액  float64
dtypes: float64(1), int64(2), object(5)
memory usage: 173.0+ MB
None
재무제표종류         0
종목코드           0
결산기준일          0
연도표시           0
분기표시           0
항목코드           0
항목명            0
분기누적금액    311741
dtype: int64


재무제표종류    종목코드       결산기준일  연도표시  분기표시  \
0  포괄별도손익계산서  060310  2015-03-31  2015     4   
1  포괄별도손익계산서  060310  2015-03-31  2015     4   
2  포괄별도손익계산서  060310  2015-03-31  2015     4   
3  포괄별도손익계산서  060310  2015-03-31  2015     4   
4  포괄별도손익계산서  060310  2015-03-31  2015     4   

                                             항목코드       항목명        분기누적금액  
0                                    ifrs_Revenue   수익(매출액)  2.110422e+10  
1                                ifrs_CostOfSales      매출원가  1.980746e+10  
2                                ifrs_GrossProfit     매출총이익  1.296757e+09  
3  dart_TotalSellingGeneralAdministrativeExpenses   판매비와관리비  5.402979e+09  
4                        dart_OperatingIncomeLoss  영업이익(손실) -4.106222e+09

## 1-6. df_Dart_BS.pkl

## 1-7. df_Bank_PL.pkl

## 1-8. df_Bank_BS.pkl